jax/_src/pjit.py

```diff
@@ -374,9 +368,48 @@ def cache_miss(*args, **kwargs):
 
   cpp_pjitted_f = wraps(fun)(cpp_pjit_f)
   cpp_pjitted_f._fun = fun
-  type(cpp_pjitted_f).clear_cache = _cpp_pjit_evict_fn
+  cpp_pjitted_f._jit_info = jit_info
+  # TODO(necula): move these to top-level; we don't need to do this for
+  # every jit
+  cpp_jitted_f_class = type(cpp_pjitted_f)
+  # TODO(necula): make clear_cache private, no need to have it part of the API
+  cpp_jitted_f_class.clear_cache = _cpp_jit_evict_fn
+  cpp_jitted_f_class.lower = _cpp_jit_lower
+  cpp_jitted_f_class.trace = _cpp_jit_trace
+  cpp_jitted_f_class.eval_shape = _cpp_jit_eval_shape
   return cpp_pjitted_f
 
+@api_boundary
+def _cpp_jit_trace(jit_func, *args, **kwargs) -> stages.Traced:
```

Comment:
Can we just call this `def trace`, `def lower` and `def eval_shape` please?

Reply:
I don't think I should call them `trace`, because it is too generic. The `_cpp_pjit_evict_fn` already used this naming scheme. Do you prefer `jit_trace`, ...?